translator/trace/otlp/otlp_to_oc_test.go

```diff
@@ -0,0 +1,172 @@
+											StringValue: &octrace.TruncatableString{Value: "eventattrval1"},
+										},
+									},
+								},
+								DroppedAttributesCount: 4,
+							},
+						},
+					},
+				},
+			},
+			DroppedMessageEventsCount: 2,
+		},
+		Attributes: &octrace.Span_Attributes{
+			DroppedAttributesCount: 1,
+		},
+		Status: &octrace.Status{Message: "status-cancelled", Code: 1},
+	}
+
+	ocSpan2 := &octrace.Span{
+		Name:      &octrace.TruncatableString{Value: "operationC"},
+		StartTime: unixnanoToTimestamp(unixnanos),
+		EndTime:   unixnanoToTimestamp(unixnanos),
+		Links: &octrace.Span_Links{
+			Link:              []*octrace.Span_Link{{}},
+			DroppedLinksCount: 1,
+		},
+	}
+
+	otlpSpan1 := &tracepb.Span{
+		Name:              "operationB",
+		StartTimeUnixnano: unixnanos,
+		EndTimeUnixnano:   unixnanos,
+		Events: []*tracepb.Span_Event{
+			{
+				TimeUnixnano: unixnanos,
+				Name:         "event1",
+				Attributes: []*commonpb.AttributeKeyValue{
+					{
+						Key:         "eventattr1",
+						Type:        commonpb.AttributeKeyValue_STRING,
+						StringValue: "eventattrval1",
+					},
+				},
+				DroppedAttributesCount: 4,
+			},
+		},
+		DroppedAttributesCount: 1,
+		DroppedEventsCount:     2,
+		Status:                 &tracepb.Status{Message: "status-cancelled", Code: tracepb.Status_Cancelled},
+	}
+
+	otlpSpan2 := &tracepb.Span{
+		Name:              "operationC",
+		StartTimeUnixnano: unixnanos,
+		EndTimeUnixnano:   unixnanos,
+		Links:             []*tracepb.Span_Link{{}},
+		DroppedLinksCount: 1,
+	}
+
+	tests := []struct {
+		name string
+		oc   consumerdata.TraceData
+		otlp tracepb.ResourceSpans
+	}{
+		{
+			name: "empty",
+			oc: consumerdata.TraceData{
+				Spans:        nil,
+				SourceFormat: sourceFormat,
+			},
+			otlp: tracepb.ResourceSpans{},
+		},
+
+		{
+			name: "nil-resource",
+			oc: consumerdata.TraceData{
+				Node:         nil,
+				Resource:     nil,
+				Spans:        nil,
+				SourceFormat: sourceFormat,
+			},
+			otlp: tracepb.ResourceSpans{
+				Resource: nil,
+			},
+		},
+
+		{
+			name: "no-spans",
+			oc: consumerdata.TraceData{
+				Node:         ocNode,
+				Resource:     ocResource,
+				Spans:        nil,
+				SourceFormat: sourceFormat,
+			},
+			otlp: tracepb.ResourceSpans{
+				Resource: &resourcepb.Resource{},
+				Spans:    []*tracepb.Span{},
+			},
+		},
+
+		{
+			name: "one-span",
+			oc: consumerdata.TraceData{
+				Node:         ocNode,
+				Resource:     ocResource,
+				Spans:        []*octrace.Span{ocSpan1},
+				SourceFormat: sourceFormat,
+			},
+			otlp: tracepb.ResourceSpans{
+				Resource: &resourcepb.Resource{},
+				Spans:    []*tracepb.Span{otlpSpan1},
+			},
+		},
+
+		{
+			name: "two-spans",
+			oc: consumerdata.TraceData{
+				Node:         ocNode,
+				Resource:     ocResource,
+				Spans:        []*octrace.Span{ocSpan1, ocSpan2},
+				SourceFormat: sourceFormat,
+			},
+			otlp: tracepb.ResourceSpans{
+				Resource: &resourcepb.Resource{},
```

Comment:
might be good to add one test with Attributes in Resource.

Reply:
Added coverage of attributes, links, tracestate etc.

Reply:
Yes, it would be nice to a bit more coverage. I agree, no need to cover internal functions specifically, since they will be rewritten soon.